llvm/lib/Target/X86/X86FastISel.cpp

```diff
@@ -2590,10 +2590,16 @@ bool X86FastISel::TryEmitSmallMemcpy(X86AddressMode DestAM,
 
 // Add an annotation to an intrinsic instruction, specifying whether the
 // intrinsic has been inlined or not.
-void annotateIntrinsic(const IntrinsicInst *II, const char *S) {
+void annotateIntrinsic(const IntrinsicInst *II, bool Inlined) {
   IntrinsicInst *CI = const_cast<IntrinsicInst *>(II);
   LLVMContext& C = CI->getContext();
-  MDNode* N = MDNode::get(C, MDString::get(C, S));
+  ConstantInt *CInt;
+  if (Inlined) {
```

Comment:
Can be:

```c
CInt = ConstantInt::get(C, APInt(1, Inlined ? 1: 0));
```

if you'd like to get rid of the `if`.

Reply:
See 25ae9c8